x-pack/plugin/esql/src/main/antlr/EsqlBaseLexer.g4

```diff
@@ -561,6 +561,10 @@ JOIN_AS : AS -> type(AS);
 JOIN_ON : ON -> type(ON), popMode, pushMode(EXPRESSION_MODE);
 USING : 'USING' -> popMode, pushMode(EXPRESSION_MODE);
 
+JOIN_UNQUOTED_SOURCE: UNQUOTED_SOURCE -> type(UNQUOTED_SOURCE);
+JOIN_QUOTED_SOURCE : QUOTED_STRING -> type(QUOTED_STRING);
```

Comment:
Thought: We _could_ make the point that currently, quoting is not required for `LOOKUP JOIN`, and we could remove this lexing rule to prohibit it.

But that'd be inconsistent and potentially confusing to a user, why they can write `FROM """test"""` but not `| LOOKUP JOIN """test""" ...`.

So I think this lexer rules make sense.

Reply:
There's at least pipes in date math that do require quoting. Example that I just ran locally and which requires quotes:
```
curl -u elastic:password -H "Content-Type: application/json" "127.0.0.1:9200/_query?format=txt" -d '
{
  "query": "from \"<test-{now/d{yyyy.MM.dd|+18:00}}>\" metadata _index"
}'
       x       |       y       |       z       |    _index     
---------------+---------------+---------------+---------------
1              |2              |2              |test-2025.01.24
``